BASE/Test/Microsoft.ApplicationInsights.Test/Microsoft.ApplicationInsights.Tests/Channel/TransmissionTest.cs

```diff
@@ -501,6 +504,224 @@ public async Task SendAsyncLogsIngestionReponseTimeAndStatusCode()
                     }
                 }
             }
+
+            [TestMethod]
+            public async Task TestTransmissionStatusEventHandlerWithSuccessTransmission()
+            {
+                // ARRANGE
+                var handler = new HandlerForFakeHttpClient
+                {
+                    InnerHandler = new HttpClientHandler(),
+                    OnSendAsync = (req, cancellationToken) =>
+                    {
+                        return Task.FromResult<HttpResponseMessage>(new HttpResponseMessage());
+                    }
+                };
+
+                using (var fakeHttpClient = new HttpClient(handler))
+                {
+                    // Instantiate Transmission with the mock HttpClient                  
+                    Transmission transmission = new Transmission(testUri, new byte[] { 1, 2, 3, 4, 5 }, fakeHttpClient, string.Empty, string.Empty);
+                    
+                    // VALIDATE
+                    transmission.TransmissionStatusEvent += delegate(object sender, TransmissionStatusEventArgs args)
+                    {
+                        Assert.AreEqual(HttpStatusCode.OK, args.Response.StatusCode);
+                    };
+
+                    // ACT
+                    HttpWebResponseWrapper result = await transmission.SendAsync();
+                }
+            }
+
+            [TestMethod]
+            public async Task TestTransmissionStatusEventHandlerWithFailureTransmission()
+            {
+                // ARRANGE
+                var handler = new HandlerForFakeHttpClient
+                {
+                    InnerHandler = new HttpClientHandler(),
+                    OnSendAsync = (req, cancellationToken) =>
+                    {
+                        return Task.FromResult<HttpResponseMessage>(new HttpResponseMessage(HttpStatusCode.ServiceUnavailable));
+                    }
+                };
+
+                using (var fakeHttpClient = new HttpClient(handler))
+                {
+                    // Instantiate Transmission with the mock HttpClient                  
+                    Transmission transmission = new Transmission(testUri, new byte[] { 1, 2, 3, 4, 5 }, fakeHttpClient, string.Empty, string.Empty);
+
+                    // VALIDATE
+                    transmission.TransmissionStatusEvent += delegate (object sender, TransmissionStatusEventArgs args)
+                    {
+                        Assert.AreEqual(HttpStatusCode.ServiceUnavailable, args.Response.StatusCode);
+                    };
+
+                    // ACT
+                    HttpWebResponseWrapper result = await transmission.SendAsync();
+                }
+            }
+
+            [TestMethod]
+            public async Task TestTransmissionStatusEventWithEventsFromMultipleIKey()
```

Comment:
we need to modify the PR description with example usage. Its not clear how would one use this. In this unit test it looks like the `Transmission` is also part of the callback. But the public API doesn't make it appear so - it only exposes `HttpResponseMessage` in the callback.

Reply:
Event handler has sender and event args. Sender object is a `Transmission` and `HttpResponseMessage` is passed as event args.